src/EnergyPlus/AirLoopHVACDOAS.cc

```diff
@@ -1018,28 +1018,28 @@ namespace AirLoopHVACDOAS {
 
     void AirLoopDOAS::GetDesignDayConditions(EnergyPlusData &state)
     {
-
-        // int constexpr summerDesignDayTypeIndex(9);
-        // int constexpr winterDesignDayTypeIndex(10);
-        for (size_t env = 1; env <= state.dataWeatherManager->Environment.size(); ++env) {
-            switch (state.dataWeatherManager->Environment(env).KindOfEnvrn) {
-            case DataGlobalConstants::KindOfSim::DesignDay:
-            case DataGlobalConstants::KindOfSim::RunPeriodDesign: {
-                if (state.dataWeatherManager->Environment(env).maxCoolingOATSizing > this->SizingCoolOATemp) {
-                    this->SizingCoolOATemp = state.dataWeatherManager->Environment(env).maxCoolingOATSizing;
+        for (auto &env : state.dataWeatherManager->Environment) {
+            if (env.KindOfEnvrn != DataGlobalConstants::KindOfSim::DesignDay && env.KindOfEnvrn != DataGlobalConstants::KindOfSim::RunPeriodDesign)
+                continue;
+            if (env.maxCoolingOATSizing > this->SizingCoolOATemp) {
+                this->SizingCoolOATemp = env.maxCoolingOATSizing;
+                if (env.KindOfEnvrn == DataGlobalConstants::KindOfSim::DesignDay &&
+                    state.dataWeatherManager->DesDayInput(env.DesignDayNum).PressureEntered) {
                     this->SizingCoolOAHumRat = Psychrometrics::PsyWFnTdpPb(
-                        state, state.dataWeatherManager->Environment(env).maxCoolingOADPSizing, DataEnvironment::StdPressureSeaLevel);
+                        state, env.maxCoolingOADPSizing, state.dataWeatherManager->DesDayInput(env.DesignDayNum).PressBarom);
```

Comment:
switcheroo on pressure used for design day moisture calculation

Reply:
Design days use the outdoor pressure value provided by the user, otherwise that input would not be needed.

    // TomorrowOutRelHum values set earlier
    state.dataWeatherManager->TomorrowOutDewPointTemp(ts, hour) =
                        Psychrometrics::PsyTdpFnWPb(state, HumidityRatio, state.dataWeatherManager->DesDayInput(EnvrnNum).PressBarom);

Reply:
Wait, the DOAS sizing is recalculating conditions across design days by looking directly at the design day inputs? Why? Seems it would be better to track peak OA conditions in the sizing routines, since they'll cycle through everything there with HumRat already calculated. Maybe that's for another pass.

And if I'm following `AirLoopDOAS::GetDesignDayConditions` correctly, it only sizes for the peak dry-bulb and coincident humidity with no check on enthalpy? So, a dewpoint design day or other cooler humid day will never be selected?

Reply:
No, the weather manager calculates the max OAT used for sizing and stores that data for each sizing environment. The AirLoopDOAS::GetDesignDayConditions` uses that data to choose the value used for sizing by searching all design days. There is not sizing routines for air loop systems. This is how it was done before and it was expanded here to include the WeatherfileDays. If there is a better way to choose how to size an outdoor air coil, say based on max enthalpy (which seems reasonable), then that code could be added. Just open an issue and suggest a course of action.

Reply:
Sorry, I was looking at the code from the prior DOAS sizing PR. Ok, this is clearly an improvement, but I will post an issue to use a better method assuming I can produce an idf that illustrates the problem.  In that issue I'll also suggest that it might be better to track the humidity ratio *after* it's calculated for the current weather in weather manager.